### PR TITLE
Domain migration DB

### PR DESCRIPTION
**Pull request by kaapstorm**
Opening this PR to work through my thinking on this.

This provides a database for undoing changes when migrating from a source Couch domain to a different destination SQL domain. When migrating, the domain of form attachment blob meta must be changed to the destination domain, and when reverting, the domain needs to be set back to its previous value. (There might be other examples of changes that I haven't yet found.)

The `Undo` model stores the original values. Please check that my indexes make sense, although that's not what's bothering me.

What I'd really like to confirm is that my router and the changes to `settings.py` make sense.

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -938,6 +938,10 @@
     ):
         database['DISABLE_SERVER_SIDE_CURSORS'] = True
 
+DATABASES['domain_migrations'] = {
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply settings.py]: Indeed.

**Comment by kaapstorm** [comment]
(force-pushed for Stickler.)

**Review by snopoke — COMMENTED** [review]
Since this is a temporary model I think it makes more sense not to have Django manage it. There's a good pattern for using SQLAlchemy with sqllite here: https://github.com/dimagi/commcare-hq/blob/3820f87f21592e710b479dc25240a480d32d5df9/corehq/apps/tzmigration/planning.py

One other comment is that you've made the model quite generic. Are there other types of undo that you're planning on tracking? If not then I'd just make it as simple as possible to fit the current use case. Especially since it's a temporary model we don't have to worry to much about changes to the schema.

(OK one more point) I know we did discuss using sqllite for some things but it seems to me that for this specific use case you could just use a CSV file with one line per `BlobMeta`. 

Note that I think to uniquely identify a `BlobMeta` object you'll need `parent_id` + `type_code` + `name` since the `id` is a per shard sequence (I guess you could use `parent_id` + `id` since that would get you to the specific shard and then the meta ID).

**Comment by kaapstorm** [comment]
That is super helpful. Thanks Simon.

Regarding why the model is so generic, BlobMeta was the first thing I found I needed to undo, but I guessed I would probably need it for other things too. Maybe not, and of course YAGNI. :upside_down_face: 

I really like the idea of just using a CSV. Much simpler! I don't need to query it; just iterating rows will be fine. I think I'll go that route.

**Pull request closed without merging**